07-complex-data-type/src/ComplexDataTypeSample.java

```diff
@@ -0,0 +1,258 @@
+import com.terracottatech.store.*;
+import com.terracottatech.store.definition.ComplexDataParseException;
+import com.terracottatech.store.definition.ComplexDataParser;
+
+import java.util.*;
+
+import static com.terracottatech.store.TypedValue.typedValue;
+
+public class ComplexDataTypeSample {
+
+    public static void main(String[] args) throws ComplexDataParseException {
+
+        /**
+         * Creating Mutable StoreList and StoreMap using newList() and newMap() methods
+         */
+
+        StoreList list = StoreList.newList()
+                .with("john").with('R')
+                .with(10)
+                .with(true)
+                .with(100101010L)
+                .with(98.6d)
+                .with(new byte[]{1, 2, 3, 4});
+
+        StoreMap map = StoreMap.newMap()
+                .with("name", "john")
+                .with("initial", 'R')
+                .with("age", 10)
+                .with("alive", true)
+                .with("marker", 100101010L)
+                .with("temp", 98.6d)
+                .with("img", new byte[]{1, 2, 3, 4});
+
+        System.out.println("StoreList is : " + list);
+        System.out.println("StoreMap is : " + map);
+        System.out.println();
+
+        /**
+         * Create a StoreMap and StoreList by parsing the String.
+         */
+
+        StoreMap mapUsingStringMapOF = StoreMap.mapOf("{ 'A' : " + 1 + ", 'F' : " + 1 + "}");
```

Comment:
Why the explicit string concatenation? Everything's a constant, and all the quoting just makes it harder to see the actual syntax.